_LowLevelCode/cpp/utility/environment.h

```diff
@@ -0,0 +1,13 @@
+#include <string>
+
+namespace Horace {
+namespace Utility {
+namespace Environment {
+// Get an environment variable
+std::string get_env_variable(const std::string &key);
+// The HORACE_ROOT environment variable
+const std::string HORACE_ROOT{get_env_variable("HORACE_ROOT")};
```

Comment:
I hadn't thought of doing it this way; I was expecting to just have a `HORACE_ROOT` defined as the constant ("HORACE_ROOT") and leaving it to users to call the function as needed.

Reply:
this has an overhead as we add more env variables and doesn't make the "failure modes" explicit
.

Reply:
Ah I see, no problem